developing_collections_migrating.rst: update (#1605)
ansible/ansible-documentation: docs/docsite/rst/dev_guide/developing_collections_migrating.rst

Migrating Ansible content to a different collection
***************************************************

> When you move content from one collection to another, for example to extract a set of related modules out of ``community.general`` or ``community.network`` to create a more focused collection, you must make sure the transition is easy for users to follow. 

Reviewer: ```suggestion
You might decide to move content from one collection to another; for example, to extract a set of related modules out of ``community.general`` or ``community.network`` to create a more focused collection.
When you migrate content between collections, you must take certain steps to ensure users can follow the transition.
```

Reviewer: I feel like this wording is a bit more to the point for the section that comes after, namely "Migrating content". When you say make sure the transition is "easy to follow" it almost sounds like optional steps to make things convenient for users.

Migrating content
=================

> If the collection you are going to migrate content from is a part of the `Ansible community package <https://github.com/ansible-community/ansible-build-data/blob/main/>`_, make sure the target collection satisfies the `Ansible collection requirements <https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html>`_. After that, to migrate the content:

Reviewer: ```suggestion
If the collection you are going to migrate content from is a part of the `Ansible community package <https://github.com/ansible-community/ansible-build-data/blob/main/>`_, ensure the target collection satisfies the `Ansible collection requirements <https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html>`_. After you satisfy the requirements, you can migrate the content as follows:
```

#. Copy content from the source (old) collection to the target (new) collection.
#. Change ``M()``, examples, ``seealso``, ``extended_documentation_fragments`` to use actual FQCNs in moved content and in other collections that have references to the content.
#. Move all related issues, pull requests, and wiki pages.

> #. Look through ``docs/docsite`` directory of `ansible-base GitHub repository <https://github.com/ansible/ansible>`_ (for example, using the ``grep`` command-line utility) to check if there are examples using the moved modules and plugins to update their FQCNs.

Reviewer: ```suggestion
#. Look through the ``docs/docsite`` directory of the `ansible-base GitHub repository <https://github.com/ansible/ansible>`_ (for example, using the ``grep`` command-line utility) to check if there are examples using the moved modules and plugins so that you can update those FQCNs.
```

#. Deprecate the module/plugin with ``removal_version`` scheduled for the next major version in ``meta/runtime.yml`` of the old collection. The deprecation must be released after the copied content has been included in a release of the new collection.
#. When the next major release of the old collection is prepared:

  * remove the module/plugin from the old collection
  * remove the symlink stored in ``plugin/modules`` directory if appropriate (mainly when removing from ``community.general`` and ``community.network``)
  * remove related unit and integration tests

>   * remove specific module utils (if they are NOT used by other modules/plugins or module_utils)

Reviewer: ```suggestion
  * remove specific module utils (if they are NOT used by other modules/plugins or ``module_utils``)
```

Migrating content
=================

> If the collection you are going to migrate content from is a part of the `Ansible community package <https://github.com/ansible-community/ansible-build-data/blob/main/>`_, ensure the target collection satisfies the `Ansible collection requirements <https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html>`_. After you satisfy the requirements, you can migrate the content as follows:

Reviewer: ```suggestion
If the collection from which you are going to migrate content is included in the `Ansible community package <https://github.com/ansible-community/ansible-build-data/blob/main/>`_, ensure the target collection satisfies the `Ansible collection requirements <https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html>`_. After you satisfy the requirements, you can migrate the content as follows:
```

> If the collection from which you are going to migrate content is included in the `Ansible community package <https://github.com/ansible-community/ansible-build-data/blob/main/>`_, ensure the target collection satisfies the `Ansible collection requirements <https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html>`_. After you satisfy the requirements, you can migrate the content as follows:

Reviewer 2: ```suggestion
If the collection from which you are going to migrate content is included in the `Ansible community package <https://github.com/ansible-community/ansible-build-data/blob/main/>`_, ensure the target collection satisfies the :ref:`collections_requirements`. After you satisfy the requirements, you can migrate the content as follows:
```

#. Copy content from the source (old) collection to the target (new) collection.
#. Change ``M()``, examples, ``seealso``, ``extended_documentation_fragments`` to use actual FQCNs in moved content and in other collections that have references to the content.
#. Move all related issues, pull requests, and wiki pages.

> #. Look through the ``docs/docsite`` directory of the `ansible-base GitHub repository <https://github.com/ansible/ansible>`_ (for example, using the ``grep`` command-line utility) to check if there are examples using the moved modules and plugins so that you can update those FQCNs.

Reviewer 2: ```suggestion
#. Look through the ``docs/docsite`` directory of the `ansible-documentation GitHub repository <https://github.com/ansible/ansible-documentation>`_ (for example, using the ``grep`` command-line utility) to check if there are examples using the moved modules and plugins so that you can update those FQCNs.
```

Reviewer 2: Should we also add an item to say check any old-collection docs/plugins for references to moved modules/plugins and update with FQCN?

Author: Mentioned the old collection in the [last commit](https://github.com/ansible/ansible-documentation/pull/1605/commits/c2fb80bdd5d585b7db54e1a8451f509854c0e59d), PTAL
